src/databricks/labs/ucx/hive_metastore/tables.scala

```diff
@@ -64,7 +64,7 @@ def metadataForAllTables(databases: Seq[String], queue: ConcurrentLinkedQueue[Ta
           }
           val isPartitioned = if (partitionColumnNames != null && !partitionColumnNames.isEmpty) true else false
 
-          Some(TableDetails("hive_metastore", databaseName, tableName, table.tableType.name, table.provider.orNull,
+          Some(TableDetails("hive_metastore", databaseName, tableName, table.tableType.name, table.provider.getOrElse("UNKNOWN"),
```

Comment:
Also make a corresponding change to TablesCrawler, which is used in integration tests

Reply:
Actually the logic implemented there comes from TablesCrawler. This part were missing